### PR TITLE
Replace scikit-build-core / CMake with setuptools

### PR DESCRIPTION
**Pull request by gmarkall**
Using scikit-build-core and CMake was overkill for copying two files into `site-packages`. This change replaces them with setuptools, and also modifies some of the package names to use a dash (i.e. `numba-cuda`) which I think is more "standard" in the places in which I've done it here.

Manual testing suggests that the packages built before and after this change are identical, with a couple of small exceptions:

- The `LICENSE` file is placed in the root by setuptools instead of a folder called `licenses`.
- Metadata differs slightly (`WHEEL`, `RECORD`, `METADATA`) due to them being generated slightly differently between the build tools.
- setuptools also includes a `top_level.txt` file.

Many thanks to @vyasr for pointing me to the solution to copy files to `site-packages` with setuptools, as used in
[`rapids-dask-dependency`](https://github.com/rapidsai/rapids-dask-dependency/).

### CODE REVIEW

File: ci/build_wheel.sh
Comment (vyasr) [inline ci/build_wheel.sh:11]: I believe that this needs to stay with an underscore. This gets used for the wheel filename, which is always normalized to have underscores even if the package name has a hyphen IIRC.

File: ci/test_wheel.sh
Comment (vyasr) [inline ci/test_wheel.sh:18]: Same as above, I think this needs to stay underscored.

File: setup.py
Comment (vyasr) [inline setup.py:14]: Nitpick: specializing the function on the filename or the path this way isn't my favorite. I would rather see this function accept two arguments, 1) the input file path relative to the package root and 2) the target directory. The latter can default to "." so that the destination defaults to `self.build_lib` and you can call the function with just the input filename below.

**Review by vyasr — CHANGES_REQUESTED** [review]

File: ci/build_wheel.sh
Comment (gmarkall) [reply ci/build_wheel.sh]: Thanks, I've undone this now.

File: ci/test_wheel.sh
Comment (gmarkall) [reply ci/test_wheel.sh]: Thanks, undone as above.

File: setup.py
Comment (gmarkall) [reply setup.py]: Changed to follow the suggestion, and verified locally that that change didn't result in a change to the built wheel.

**Review by vyasr — APPROVED** [review]
My review has no power here, but this LGTM.

**Comment by gmarkall** [comment]
> My review has no power here, but this LGTM.

On the contrary, I would not merge it without your approval :slightly_smiling_face:

**Pull request merged**
